sound/soc/sof/sof-priv.h

```diff
@@ -155,6 +155,14 @@ struct snd_sof_dsp_ops {
 			   enum snd_sof_fw_blk_type type, u32 offset,
 			   void *src, size_t size); /* mandatory */
 
+	/* Mailbox IO */
+	void (*mailbox_read)(struct snd_sof_dev *sof_dev,
+			     u32 offset, void *dest,
+			     size_t size); /* mandatory */
+	void (*mailbox_write)(struct snd_sof_dev *sof_dev,
+			      u32 offset, void *src,
+			      size_t size); /* mandatory */
+
```

Comment:
@Potochi if we make mailbox_read /  mailbox_write optional as AMD guys suggested please modify the comment here to read 'optional' instead of 'mandatory'

Reply:
@ajitkupandey can you point us to the AMD implementation of ipc_msg_data? We want to understand if you can use the generic functions introduced by our commit. 

We use the abstraction of the 'mailbox' to exchange various information between DSP and Application Processor. You don't need to have IO mapped memory, it can be everything.

Reply:
@dbaluta  yeah we are also using a kind of mailbox mechanism but not implemented mailbox_read()/write() .. you can refer to for ipc_msg_data() implementation https://github.com/thesofproject/linux/pull/3054/commits/1f96f249e1b26b280fa9d0122392e6cf92965ded#diff-1d060dee752ac0ca2adf00c0754be0bf9b554767e4585b6c9cbafc224a94114dR152

Reply:
Fixed.